### DISCUSSION TITLE
Production VS Staging with different table entries

### DISCUSSION THREAD

**Discussion by marksie1988** [theory/sqitch#682] [Q&A]
Hi All, 

I have a production and staging environment, in both there is a table `cron_jobs` however the cron jobs have different settings if they are in production vs staging. 

e.g. one job may run on Tuesday on production but Wednesday in staging.  or the job might be disabled in staging and enabled in production.

Each cron job has a single line entry with its settings which means I need a way to say:

1. on production, insert these entries
2. on staging, insert these different entries

My question is, how would I achieve this?

**Comment by theory** [comment]
Can you use [variables](http://sqitch.org/docs/manual/sqitch-deploy/#configuration-variables)? Something like:

```
sqitch target alter prd --set cron_entries='5 4 * * * /foo/bar'
sqitch target alter stg --set cron_entries='* * * * * /foo/bar'
```

Then use the variable for your engine. For Postgres, it'd be something like:

``` sql
INSERT INTO cron_jobs VALUES (:'cron_entries');
```

Another option is to set a variable for your environment then write a conditional in the deploy script. Something like:

```
sqitch target alter prd --set env=prd
sqitch target alter stg --set env=stg
```

Then in the deploy script, do something like this (assuming Postgres here, just typed in, not tested):

``` sql
DO $$
DECLARE env TEXT := $$ || :'env' || $$
BEGIN
  IF env = 'prd' THEN
    INSERT INTO cron_jobs (prd stuff);
  ELSE
    INSERT INTO cron_jobs (stg stuff);
  END IF;
END;
$$;
